src/test/java/mate/academy/dao/impl/RoleDaoImplTest.java

```diff
@@ -0,0 +1,40 @@
+package mate.academy.dao.impl;
+
+import java.util.Optional;
+import mate.academy.AbstractTest;
+import mate.academy.dao.RoleDao;
+import mate.academy.model.Role;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+class RoleDaoImplTest extends AbstractTest {
+    private RoleDao roleDao;
+    private Role role;
+
+    @Override
+    protected Class<?>[] entities() {
+        return new Class[]{Role.class, Role.RoleName.class};
+    }
+
+    @BeforeEach
+    void setUp() {
+        roleDao = new RoleDaoImpl(getSessionFactory());
+        role = new Role(Role.RoleName.USER);
+    }
+
+    @Test
+    void save_OK() {
```

Comment:
let's add a test for the case if you try to add a role with a name that already exists in the database